src/psyclone/psyad/domain/lfric/lfric_adjoint_harness.py

```diff
@@ -365,32 +426,70 @@ def generate_lfric_adjoint_harness(tl_psyir):
     kern = lfalg.kernel_from_metadata(parse_tree, kernel_name)
     kern_args = lfalg.construct_kernel_args(routine, kern)
 
+    # Validate the index values for the coordinate and face_id fields if
+    # supplied.
+    geometry_arg_indices = []
+    if coord_arg_idx is not None:
+        _validate_geom_arg(kern, coord_arg_idx, "coordinate", ["wchi"], 3)
```

Comment:
I wasn't sure about this but I've done it. Really, it would be better if the LFRic metadata followed GOcean and allowed the fact that a kernel needs these properties to be specified separately.